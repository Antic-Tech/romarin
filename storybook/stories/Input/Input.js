import React from 'react';
import { TextInput } from 'react-native';
import PropTypes from 'prop-types';
import ThemeProvider from '../../theme';

const Input = props => {
	return (
		<TextInput
			style={{
				borderRadius: ThemeProvider.theme.inputBorderRadius,
				padding: ThemeProvider.theme.inputPadding,
				margin: ThemeProvider.theme.inputMargin,
				borderColor: ThemeProvider.theme.inputBorderColor,
				borderWidth: ThemeProvider.theme.inputBorderWidth,
				flexDirection: 'row'
			}}
			{...props}
		/>
	);
};

Input.propTypes = {
	size: PropTypes.string,
	muted: PropTypes.bool
};

Input.defaultProps = {
	size: 'medium',
	muted: false
};

export { Input as default };
