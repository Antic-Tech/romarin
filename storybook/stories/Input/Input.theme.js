export default core => ({
	inputBorderRadius: core.coreBorderRadius,
	inputPadding: core.corePaddingSmall,
	inputMargin: core.corePaddingSmall,
	inputBorderColor: core.colorBlack,
	inputBorderWidth: 1
});
