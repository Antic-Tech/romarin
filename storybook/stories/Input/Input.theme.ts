export default (core) => ({
	inputBorderRadius: core.coreBorderRadius,
	inputPadding: core.corePaddingSmall,
	inputPaddingHorizontal: core.corePaddingSmall,
	inputMargin: core.corePaddingSmall,
	inputLabelMargin: core.corePaddingXSmall,
	inputBorderColor: core.colorLighterGray,
	focusedInputBorderColor: core.colorPrimary,
	focusedInputLabelColor: core.colorPrimary,
	inputLabelColor: core.colorPrimary,
	inputTextColor: core.colorPrimary,
	inputBorderWidth: 1
});
