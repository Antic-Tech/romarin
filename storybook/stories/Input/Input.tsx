import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { TextInput, View } from 'react-native';
import { Text } from '../../components';
import { withTheme } from '../../theme';

class Input extends Component {
	constructor(props) {
		super(props);

		this.state = {
			isFocused: false
		};
		this.onBlur = this.onBlur.bind(this);
		this.onFocus = this.onFocus.bind(this);
	}

	onFocus(event) {
		this.setState({ isFocused: true });

		if (this.props.onFocus) {
			this.props.onFocus(event);
		}
	}

	onBlur() {
		this.setState({ isFocused: false });

		if (this.props.onBlur) {
			this.props.onBlur();
		}
	}

	render() {
		const { label, theme, ...props } = this.props;
		const { isFocused } = this.state;

		return (
			<View
				style={{
					borderRadius: theme.inputBorderRadius,
					borderColor: isFocused
						? theme.focusedInputBorderColor
						: theme.inputBorderColor,
					backgroundColor: theme.inputBackgroundColor,
					margin: theme.inputMargin,
					padding: theme.inputPadding,
					paddingHorizontal: theme.inputPaddingHorizontal,
					borderWidth: theme.inputBorderWidth,
					flexDirection: 'row',

					alignItems: 'center'
				}}
			>
				<Text
					style={{
						width: 80,
						margin: theme.inputLabelMargin,
						color: isFocused
							? theme.focusedInputLabelColor
							: theme.inputLabelColor,
						fontSize: theme.coreFontSizeMedium
						// textAlign: 'right'
					}}
					weight="bold"
				>
					{label}
				</Text>
				<TextInput
					style={{
						flexDirection: 'row',
						flex: 1,
						fontSize: theme.coreFontSizeMedium,
						color: theme.inputTextColor
					}}
					selectionColor={theme.focusedInputLabelColor}
					placeholderTextColor={theme.inputPlaceholderColor}
					onFocus={this.onFocus}
					onBlur={this.onBlur}
					{...props}
				/>
			</View>
		);
	}
}

Input.propTypes = {
	size: PropTypes.string,
	muted: PropTypes.bool
};

Input.defaultProps = {
	size: 'medium',
	muted: false,
	label: ''
};

export default withTheme(Input)

