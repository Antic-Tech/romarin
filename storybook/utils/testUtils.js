import { mount, shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

export const renderComponent = (jsx, { shallowRender = true, store } = {}) => {
	const renderOptions = { context: { store } };

	if (shallowRender) {
		return shallow(jsx, renderOptions);
	}

	return mount(jsx, renderOptions);
};

export const componentFactory = Component => (props = {}, options = {}) => {
	return renderComponent(<Component {...props} />, options);
};

export const getTree = jsx => renderer.create(jsx).toJSON();

export const verifySnapshot = jsx => {
	const tree = getTree(jsx);

	expect(tree).toMatchSnapshot();
};
