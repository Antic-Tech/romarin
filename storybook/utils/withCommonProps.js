import React from 'react';
import { StyleSheet } from 'react-native';
import ThemeProvider from '../theme';

const Sizes = {
	XXSmall: 'XXSmall',
	XSmall: 'XSmall',
	Small: 'Small',
	Medium: 'Medium',
	Large: 'Large',
	XLarge: 'XLarge',
	XXLarge: 'XXLarge'
};

const withPadding = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				padding,
				paddingHorizontal,
				paddingVertical,
				paddingTop,
				paddingBottom,
				paddingLeft,
				paddingRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					padding: padding ? ThemeProvider.theme[`corePadding${padding}`] : undefined,
					paddingHorizontal: paddingHorizontal
						? ThemeProvider.theme[`corePadding${paddingHorizontal}`]
						: undefined,
					paddingVertical: paddingVertical ? ThemeProvider.theme[`corePadding${paddingVertical}`] : undefined,
					paddingTop: paddingTop ? ThemeProvider.theme[`corePadding${paddingTop}`] : undefined,
					paddingBottom: paddingBottom ? ThemeProvider.theme[`corePadding${paddingBottom}`] : undefined,
					paddingLeft: paddingLeft ? ThemeProvider.theme[`corePadding${paddingLeft}`] : undefined,
					paddingRight: paddingRight ? ThemeProvider.theme[`corePadding${paddingRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withMargin = (Component) =>
	class WithLoading extends React.Component {
		render() {
			const {
				style,
				margin,
				marginHorizontal,
				marginVertical,
				marginTop,
				marginBottom,
				marginLeft,
				marginRight,
				...props
			} = this.props;

			const newStyles = StyleSheet.flatten([
				{
					margin: margin ? ThemeProvider.theme[`coreMargin${margin}`] : undefined,
					marginHorizontal: marginHorizontal
						? ThemeProvider.theme[`coreMargin${marginHorizontal}`]
						: undefined,
					marginVertical: marginVertical ? ThemeProvider.theme[`coreMargin${marginVertical}`] : undefined,
					marginTop: marginTop ? ThemeProvider.theme[`coreMargin${marginTop}`] : undefined,
					marginBottom: marginBottom ? ThemeProvider.theme[`coreMargin${marginBottom}`] : undefined,
					marginLeft: marginLeft ? ThemeProvider.theme[`coreMargin${marginLeft}`] : undefined,
					marginRight: marginRight ? ThemeProvider.theme[`coreMargin${marginRight}`] : undefined
				},
				style
			]);

			return <Component style={newStyles} {...props} />;
		}
	};

const withCommonProps = (Component) => withPadding(withMargin(Component));

export { withPadding, withMargin, withCommonProps };
