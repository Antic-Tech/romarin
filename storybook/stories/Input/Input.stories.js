import React from 'react';
import { View } from 'react-native';
import { storiesOf } from '@storybook/react-native';
import { action } from '@storybook/addon-actions';

import Input from './Input';
import CenterView from '../CenterView';

storiesOf('Input', module)
	.addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
	.add('default', () => (
		<React.Fragment>
			<Input placeholder="Email" />
			<Input placeholder="Password" />

			<Input placeholder="Input" />

			<Input placeholder="Input" />

			<Input placeholder="Input" />
		</React.Fragment>
	));
