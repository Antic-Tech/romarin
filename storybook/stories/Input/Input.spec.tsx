import React from 'react';

import Input from './Input';

import { verifySnapshot } from '../../utils/testUtils';

describe('Input', () => {
	describe('Snapshots', () => {
		it('renders correctly', () => {
			verifySnapshot(<Input />);
		});
	});
});
