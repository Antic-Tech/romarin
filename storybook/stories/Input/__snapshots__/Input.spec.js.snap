// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshots renders correctly 1`] = `
<TextInput
  allowFontScaling={true}
  muted={false}
  size="medium"
  style={
    Object {
      "borderColor": "#000000",
      "borderRadius": 10,
      "borderWidth": 1,
      "flexDirection": "row",
      "margin": 8,
      "padding": 8,
    }
  }
/>
`;
